test-suite/js/src/test/scala/org/scalajs/testsuite/compiler/InstanceTestsHijackedBoxedClassesTest.scala

```diff
@@ -84,7 +84,10 @@ class InstanceTestsHijackedBoxedClassesTest {
 
   @Test def should_support_asInstanceOf_negative(): Unit = {
     assumeTrue("Assumed compliant asInstanceOf", hasCompliantAsInstanceOfs)
-    assertThrows(classOf[Exception], (12345: Any).asInstanceOf[Unit])
+    if (!scalaVersion.startsWith("2.12."))
+      assertThrows(classOf[Exception], (12345: Any).asInstanceOf[Unit])
+    else
+      (12345: Any).asInstanceOf[Unit]
```

Comment:
The `else` part should be in `should_support_asInstanceOf_positive()` (with an `if (scalaVersion.startsWith("2.12.")))`), because it must also work when we do not have compliant as-instance-ofs.